### PR TITLE
#245 refactor transfer module (fee income)

### PR DESCRIPTION
**Pull request by wsbrenk**
1. changes proposed in this pull request:

   - fixes issue #245, #1938
   - if not fixing an existing issue, comments explaining the purpose of the PR should be provided)

2. `src/main/resources/release_notes.md` ...

 - [x] has been updated
 - [ ] does not require update

3. [Optional] suggested person to review this PR @tychobrailleur

### CODE REVIEW

File: src/main/java/module/transfer/history/HistoryPane.java [hunk 1]
```diff
@@ -217,7 +210,7 @@ public HistoryPane() {
         setLeftComponent(topPanel);
         setRightComponent(playerDetailPanel);
 
-        refresh();
+        //refresh();
```
Comment (tychobrailleur) [inline src/main/java/module/transfer/history/HistoryPane.java:213]: Is that intended?

File: src/main/java/module/transfer/history/PlayerTransferTableModel.java [hunk 2]
```diff
@@ -1,89 +1,178 @@
...
-            case 4 -> PlayerTransfer.BUY;
-            case 5 -> transfer.getSellerName();
-            case 6 -> transfer.getPrice();
-            case 7 -> transfer.getTsi();
-            case 8 -> transfer.getPlayerInfo()!=null?transfer.getPlayerInfo().getAgeWithDaysAsString(transfer.getDate()):"";
-            case 9 -> new JButton(HOVerwaltung.instance().getLanguageString("ls.button.delete"));
-            default -> ""; //$NON-NLS-1$
-        };
+//
+//    //~ Methods ------------------------------------------------------------------------------------
```
Comment (tychobrailleur) [inline src/main/java/module/transfer/history/PlayerTransferTableModel.java:119]: Just clean up all the commented out stuff. :-)

File: src/main/java/module/transfer/history/TransferTableModel.java [hunk 3]
```diff
@@ -1,213 +1,265 @@
...
-                if (transfer.getPlayerInfo() != null) {
-                    return transfer.getPlayerInfo().getSPskill();
-                } else {
-                    return -1;
-                }
-
-            default:
-                return ""; //$NON-NLS-1$
+    @Override
+    protected void initData() {
```
Comment (tychobrailleur) [inline src/main/java/module/transfer/history/TransferTableModel.java:251]: Interestingly I have been looking into migrating these this morning, and was really thinking I should look into removing all the duplication in that space.  On my todo!

**Review by tychobrailleur — COMMENTED** [review]
At a 20,000-foot level, this seems good!  Looking forward to giving this a try!

File: src/main/java/module/transfer/XMLParser.java [hunk 4]
```diff
@@ -114,86 +110,69 @@ public static List<PlayerTransfer> getAllPlayerTransfers(int playerId) {
...
-	        List<PlayerTransfer> transfers = parseTeamTransfers(doc, activatedDate, endDate);
-	        if (transfers.size()<1) {
-	        	stop = true;
+        while (true) {
+            var transfers = downloadTeamTransfers(teamId, page++);
+	        if (transfers.isEmpty()) {
+                break;
 	        } else {
-	        	transferList.addAll(transfers);
+                for ( var transfer : transfers){
```
[the remark this answers is not in the document]
Comment (wsbrenk) [reply src/main/java/module/transfer/XMLParser.java:125]: I don't think so. `transfers` contains maximum 25 transfers downloaded page by page from chpp.  If this page is empty, the loop is exited. These pages are added to the return value `transferList`.

File: src/main/java/core/db/AbstractTable.java
[the remark this answers is not in the document]
Comment (wsbrenk) [reply src/main/java/core/db/AbstractTable.java:1]: I'm looking forward to the result

File: src/main/java/module/transfer/history/HistoryPane.java (hunk 1 above)
Comment (wsbrenk) [reply src/main/java/module/transfer/history/HistoryPane.java:213]: Yes - this refresh is again called from outside and it costs double cpu workload.

File: src/main/java/module/transfer/history/PlayerTransferTableModel.java (hunk 2 above)
Comment (wsbrenk) [reply src/main/java/module/transfer/history/PlayerTransferTableModel.java:119]: done

File: src/main/java/module/transfer/XMLParser.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (tychobrailleur) [reply src/main/java/module/transfer/XMLParser.java:125]: Oh got it, it iterates through the pages, until it finally reaches an empty list and breaks out of the loop.

**Pull request merged**